### ISSUE TITLE
[HOLD for payment 2024-05-06] [$250][Search v1] Add ReportScreen to RHP

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#39883]
Section from [design doc](https://docs.google.com/document/d/1u6tBxmoKAEc70jvyZHaKykcfyC5iyEaewfgYXUSSbBc/edit#bookmark=id.7n9gmhyf70r4)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017622f24b31111163</li>
        <li>Upwork Job ID: 1780987474836803584</li>
        <li>Last Price Increase: 2024-04-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @adelekennedy</details>

**Comment by WojtekBoman** [comment]
Hi! I'm going to work on that :)

**Comment by WojtekBoman** [comment]
@luacmartins 
I'd like to confirm that this is the expected behaviour when we open a thread from ReportScreen in RHP:

https://github.com/Expensify/App/assets/47774969/9642a71e-0278-41c5-b21f-21026056f83b

**Comment by luacmartins** [comment]
I'd love @Expensify/design opinion on whether other links clicked in the RHP ReportScreen should also open in the RHP, but I'd say let's ship the MVP the way you have it and polish this later, since this is probably not trivial and would require a decent amount of work to be implemented.

**Comment by shawnborton** [comment]
I'd say we should bring this discussion to Slack, but I love the idea of us opening up search results in the RHP. I could see us later adding in some way to like "expand" the view into the main pane (which would just take you to the Inbox/Chats page).

**Comment by luacmartins** [comment]
@shawnborton clicking on a search result row will open the result in the RHP as described in the HL design doc for Search. Now the question is if we wanna keep any navigation that happens in the RHP contained in the RHP, e.g. should clicking on a link to a task open the task in the RHP or main pane?

**Comment by shawnborton** [comment]
Ah, got it. Hmm, good question... I kind of think you'd be able to keep using the app within the RHP itself. Like let's say you open a report from Search in the RHP. Maybe you then want to open up one of the expenses, and then even edit one of the expenses. So part of me thinks we'd just let you stay in the RHP as long as you'd like? 

Curious for what @Expensify/design @trjExpensify @JmillsExpensify thinks though.

**Comment by dannymcclain** [comment]
Yeah I agree with Shawn—I think we should keep it in the RHP. To me, the reason why is basically about maintaining your context on the search page. Say I'm going down a list of reports/expenses I need to deal with, I find one that I want to go into to make a couple changes or leave a few comments, but when I'm done with that one I want to continue going down my list the search page.

We're also talking about including some way to "expand" or "navigate to the original" for all these items that are opened in the RHP. So if we had that for every chat/report/task/etc that was opened in the RHP, you'd always have an easy way to go to the "full" version of the thing you're looking at.

**Comment by luacmartins** [comment]
@WojtekBoman I suspect that keeping every link in the RHP context will involve quite a few changes to the navigation logic, so I suggest we break this down into a separate issue and tackle it once the basic search page is in place.

**Comment by adelekennedy** [comment]
@hoangzinh sorry for the delay! I sent you an offer on Upwork!

**Comment by adelekennedy** [comment]
Payouts due:

- [ ] Contributor: $250 @hoangzinh 

Upwork job is [here](https://www.upwork.com/ab/applicants/1780987474836803584/hired).

**Comment by adelekennedy** [comment]
Done!
